handshake/interface.go

```diff
@@ -0,0 +1,19 @@
+package handshake
+
+import "github.com/lucas-clemente/quic-go/protocol"
+
+// Sealer seals a packet
+type Sealer func(dst, src []byte, packetNumber protocol.PacketNumber, associatedData []byte) []byte
```

Comment:
I think I would prefer to define a Sealer interface with a single Seal function. Then we can just return the implementations of the various Sealers directly. If we need a lambda somewhere (although I don't see any place right now), we can to something similar to how https://golang.org/pkg/net/http/#HandlerFunc works. What to you think?